src/pages/workspaces/workspace/analysis/Analyses.test.ts

```diff
@@ -0,0 +1,389 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import { Ajax } from 'src/libs/ajax'
+import { isFeaturePreviewEnabled } from 'src/libs/feature-previews'
+import { ENABLE_JUPYTERLAB_ID, JUPYTERLAB_GCP_FEATURE_ID } from 'src/libs/feature-previews-config'
+import { goToPath } from 'src/libs/nav'
+import { getLocalPref, setLocalPref } from 'src/libs/prefs'
+import LoadedState from 'src/libs/type-utils/LoadedState'
+import {
+  defaultAzureWorkspace,
+  defaultGoogleWorkspace
+} from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import {
+  AnalysesData,
+  AnalysesProps,
+  BaseAnalyses, getUniqueFileName
+} from 'src/pages/workspaces/workspace/analysis/Analyses'
+import {
+  AbsolutePath, FileName, findPotentialNotebookLockers, notebookLockHash
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import { analysisLauncherTabName } from 'src/pages/workspaces/workspace/analysis/runtime-common-components'
+import { toolLabels } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { AnalysisFile, getFileFromPath, useAnalysisFiles } from 'src/pages/workspaces/workspace/analysis/useAnalysisFiles'
+import { asMockedFn } from 'src/testing/test-utils'
+
+
+type NavExports = typeof import('src/libs/nav')
+jest.mock('src/libs/nav', (): NavExports => {
+  const originalModule = jest.requireActual('src/libs/nav')
+  return {
+    ...originalModule,
+    getLink: jest.fn(),
+    goToPath: jest.fn()
+  }
+})
```

Comment:
Very nitpicky, so feel free to ignore... these could be formatted differently to make them a bit more concise.

```suggestion
jest.mock('src/libs/nav', (): NavExports => ({
  ...jest.requireActual('src/libs/nav'),
  getLink: jest.fn(),
  goToPath: jest.fn(),
}))
```

Reply:
🤷 I do like this more concise approach 👍